src/interrupts.c

```diff
@@ -0,0 +1,46 @@
+#include "interrupts.h"
+#include "libixy-vfio.h"
+#include "log.h"
+#include "stats.h"
+
+#include <stdio.h>
+
+/**
+ * Calculate packets per second based on the received number of packets and the elapsed time in nanoseconds since the
+ * last calculation.
+ * @param received_pkts Number of received packets.
+ * @param elapsed_time_nanos Time elapsed in nanoseconds since the last calculation.
+ * @return Packets per second.
+ */
+static double mpps(uint64_t received_pkts, uint64_t elapsed_time_nanos) {
+	return (double) received_pkts / 1000000.0 / ((double) elapsed_time_nanos / 1000000000.0);
+}
```

Comment:
can this be done without using floating point?

Reply:
Because of performance? Should I also change the average moving then to integers only? I will try to convert it to pps instead of mpps :)

Reply:
ok I changed the code now to work without floating points :)